parachains/runtimes/assets/asset-hub-kusama/src/xcm_config.rs

```diff
@@ -471,7 +480,11 @@ type LocalXcmRouter = (
 
 /// The means for routing XCM messages which are not for local execution into the right message
 /// queues.
-pub type XcmRouter = WithUniqueTopic<(LocalXcmRouter, bridging::BridgingXcmRouter)>;
+pub type XcmRouter = WithUniqueTopic<(
+	LocalXcmRouter,
+	// Router which wraps and sends xcm to BridgeHub to be delivered to the Polkadot GlobalConsensus
+	ToPolkadotXcmRouter,
```

Comment:
@bkontur Just some random thought. Can you, please, confirm that noone can use the XCM pallet send (or something like that) to send the program with `ExportMessage` instruction from AH to sibling BH? Otherwise he could avoid paying the proper fee for that.

Reply:
@svyatonik do you mean something like if any other sibling parachain send `xcm::Transact(pallet_xcm::send(to_bridge_hub, ExportMessage(...)))` ?

all AssetHubs have disabled `pallet_xcm::send`:
```
        // We want to disallow users sending (arbitrary) XCMs from this chain.
	type SendXcmOrigin = EnsureXcmOrigin<RuntimeOrigin, ()>;
...
       type XcmExecuteFilter = Nothing;
```

but I think I got your good point,
if above stuff will be allowed, we would charge origin by `XcmpQueue` router fee and not by `ToPolkadotXcmRouter` fee, if XcmpQueue fee will be enough and lower than `ToPolkadotXcmRouter`, then yes we could have a problem



I think also there is a simple solution for this:
`pallet_xcm::send_xcm` [adds `DescendOrigin`](https://github.com/paritytech/polkadot/blob/master/xcm/pallet-xcm/src/lib.rs#L1516-L1528), so if we adjust xcm filter on BridgeHubs e.g. do not allow `DescendOrigin` or allow only `MultiLocation {parents: 1, interior: X1(Parachain(para_id))}` pattern it should be enough

Reply:
I trust your judgement here :) If we do not allow sending arbitrary messages to sibling BH **now**, then probably it is OK - just wanted to be sure. For v2 we won't need it (I assume), because we will be accepting messages from non-system parachains and (I guess) we don't want to force them to use something that is not strictly necessary (like disallowing programs with `DescendOrigin` instruction). Because in any case, we'll either need to support fishermens, or impl something like https://github.com/paritytech/parity-bridges-common/issues/2324 for activating backpressure